manifest_staging/charts/secrets-store-csi-driver/templates/crds-upgrade-hook.yaml

```diff
@@ -36,31 +38,34 @@ metadata:
 {{ include "sscd.labels" . | indent 2 }}
   annotations:
     helm.sh/hook: pre-install,pre-upgrade
-    helm.sh/hook-delete-policy: before-hook-creation,hook-succeeded
+    helm.sh/hook-delete-policy: "hook-succeeded,before-hook-creation"
+    helm.sh/hook-weight: "1"
 ---
-apiVersion: v1
-kind: Pod
+apiVersion: batch/v1
+kind: Job
 metadata:
   name: {{ template "sscd.fullname" . }}-upgrade-crds
   namespace: {{ .Release.Namespace }}
-  annotations:
-    helm.sh/hook: pre-install,pre-upgrade
-    helm.sh/hook-delete-policy: before-hook-creation,hook-succeeded
 {{ include "sscd.labels" . | indent 2 }}
+  annotations:
+    helm.sh/hook: pre-upgrade
+    helm.sh/hook-weight: "1"
+    helm.sh/hook-delete-policy: "hook-succeeded,before-hook-creation"
 spec:
-  serviceAccountName: {{ template "sscd.fullname" . }}-upgrade-crds
-  restartPolicy: OnFailure
-  containers:
-    - name: crds-upgrade
-      image: "{{ .Values.linux.crds.image.repository }}:{{ .Values.linux.crds.image.tag }}"
-      args:
-      - apply
-      - -f
-      - crds/
-      imagePullPolicy: {{ .Values.linux.crds.image.pullPolicy }}
-  {{- if .Values.imagePullSecrets }}
-  imagePullSecrets:
-    {{ toYaml .Values.imagePullSecrets | indent 4 }}
-  {{- end }}
-  nodeSelector:
-    kubernetes.io/os: linux
+  backoffLimit: 0
+  template:
+    metadata:
+      name: crds-keep
```

Comment:
Isn't this install/upgrade hook? Also can we uniform naming with `{{ template "sscd.fullname" . }}` prefix like other names?

Reply:
this is only required during upgrade from the current charts `0.0.23` to newer versions. For fresh install, crds are installed by helm.